zone/questmgr.cpp

```diff
@@ -359,22 +359,25 @@ Mob *QuestManager::spawn_from_spawn2(uint32 spawn2_id)
 
 void QuestManager::enable_spawn2(uint32 spawn2_id)
 {
-	database.UpdateSpawn2Status(spawn2_id, 1);
+	database.UpdateSpawn2Status(spawn2_id, 1, zone->GetInstanceID());
 	auto pack = new ServerPacket(ServerOP_SpawnStatusChange, sizeof(ServerSpawnStatusChange_Struct));
-	ServerSpawnStatusChange_Struct* ssc = (ServerSpawnStatusChange_Struct*) pack->pBuffer;
-	ssc->id = spawn2_id;
-	ssc->new_status = 1;
+	auto *ssc = (ServerSpawnStatusChange_Struct *) pack->pBuffer;
+	ssc->id          = spawn2_id;
+	ssc->new_status  = true;
+	ssc->instance_id = zone->GetInstanceID();
```

Comment:
Not sure if any current usage of these functions are called from outside the spawn2 id's zone but this could end up using the wrong instance id if so. Otherwise since this change ties these functions to the current zone's instance then there should be no need to send a world message to other zones and the spawns could just be toggled immediately.

This isn't super relevant for this pr but I think this needs to be looked at closer and the intent of the api decided in a followup to either add an instance id overload or remove a possible unnecessary world message.

Reply:
Yeah I’ll look at it closer later

Reply:
Looking into this code, its actually really strange as it is meant only for that zone - yet it doesn't filter by zone id so you could technically end up toggling another zone's spawns.

Going to localize this to a function

Reply:
Updated, please review result

Reply:
Actually I reverted it entirely. After thinking about it, there was a reason for the way it was done originally. 

If you wanted to disable spawns from another zone based on events that occurred in another, the code enables you to do so. Although, the current `quest::` `eq.` variants do assume current `instance_id` which in most of the use cases today being in normal zones, this won't be a problem.

If someone needs to add functionality for cross zone targetting. The API's could be added